src/zampy/datasets/land_cover.py

```diff
@@ -0,0 +1,249 @@
+            download_dir,
+            time_bounds,
+            spatial_bounds,
+            variable_names,
+        )
+
+        download_folder = download_dir / self.name
+        download_folder.mkdir(parents=True, exist_ok=True)
+
+        cds_utils.cds_request_land_cover(
+            dataset=self.cds_dataset,
+            time_bounds=time_bounds,
+            path=download_folder,
+            overwrite=overwrite,
+        )
+
+        write_properties_file(
+            download_folder, spatial_bounds, time_bounds, variable_names
+        )
+
+        return True
+
+    def ingest(
+        self,
+        download_dir: Path,
+        ingest_dir: Path,
+        overwrite: bool = False,
+    ) -> bool:
+        download_folder = download_dir / self.name
+        ingest_folder = ingest_dir / self.name
+        ingest_folder.mkdir(parents=True, exist_ok=True)
+
+        archive_file_pattern = f"{self.name}_*.zip"
+        archive_files = list(download_folder.glob(archive_file_pattern))
+
+        for file in archive_files:
+            unzip_raw_to_netcdf(
+                ingest_folder,
+                file=file,
+                overwrite=overwrite,
+            )
+
+        copy_properties_file(download_folder, ingest_folder)
+
+        return True
+
+    def load(
+        self,
+        ingest_dir: Path,
+        time_bounds: TimeBounds,
+        spatial_bounds: SpatialBounds,
+        resolution: float,
+        regrid_method: str,  # Unused in land-cover dataset
+        variable_names: list[str],
+    ) -> xr.Dataset:
+        files: list[Path] = []
+        for var in variable_names:
+            if var not in self.variable_names:
+                msg = (
+                    "One or more variables are not in this dataset.\n"
+                    f"Please check input. Dataset: '{self.name}'\n"
+                    f"Variables: '{variable_names}'"
+                )
+                raise ValueError(msg)
+        files = list((ingest_dir / self.name).glob(f"{self.name}_*.nc"))
+
+        ds = xr.open_mfdataset(files, chunks={"latitude": 200, "longitude": 200})
+        ds = ds.sel(time=slice(time_bounds.start, time_bounds.end))
+        new_grid = xarray_regrid.Grid(
+            north=spatial_bounds.north,
+            east=spatial_bounds.east,
+            south=spatial_bounds.south,
+            west=spatial_bounds.west,
+            resolution_lat=resolution,
+            resolution_lon=resolution,
+        )
+        target_dataset = xarray_regrid.create_regridding_dataset(new_grid)
+
+        ds_regrid = ds.regrid.most_common(target_dataset, time_dim="time", max_mem=1e9)
+
+        return ds_regrid
+
+    def convert(
+        self,
+        ingest_dir: Path,
+        convention: str | Path,
+    ) -> bool:
+        converter.check_convention(convention)
+        ingest_folder = ingest_dir / self.name
+
+        data_file_pattern = "land-cover_LCCS_MAP_*.nc"
+
+        data_files = list(ingest_folder.glob(data_file_pattern))
```

Comment:
Is there a better way to look for files? Here the data file pattern is hard coded and very hidden.

Reply:
The file name pattern is decided by the data provider, so I think it's fine to hard code it.

Reply:
Thanks for your comment. The fixed data pattern comes from the downloading, where we name the downloaded data like this:
https://github.com/EcoExtreML/zampy/blob/703a01c48b1bf03f07227ef763f26ea8e169936e/src/zampy/datasets/cds_utils.py#L138

Again, this comes back to the discussion about how do we expect users to use this tool. Since for now, we don't want our users to call these functions (or even see these APIs), but simply stick to the recipe (https://github.com/EcoExtreML/zampy/blob/main/docs/using_zampy.md), it is good to fix the name patterns. Let's discuss it in our next meeting.

Reply:
Issue #37 is relevant here as well.